packages/backend-defaults/src/entrypoints/cache/CacheManager.ts

```diff
@@ -109,42 +106,200 @@ export class CacheManager {
+        storeConfigPath,
+        config,
+        logger,
+      );
+    }
     return undefined;
   }
 
-  /**
-   * Parse Redis-specific options from configuration.
-   */
+  private static parseInfinispanOptions(
+    storeConfigPath: string,
+    config: RootConfigService,
+    logger?: LoggerService,
+  ): InfinispanCacheStoreOptions | undefined {
+    const infinispanConfig = config.getConfig(storeConfigPath);
+    const parsedOptions: Partial<InfinispanCacheStoreOptions> = {};
+
+    if (infinispanConfig.has('servers')) {
+      const serversConfig = infinispanConfig.get('servers');
+      if (Array.isArray(serversConfig)) {
+        parsedOptions.servers = infinispanConfig.getConfigArray('servers').map(
+          serverConf =>
+            ({
+              host: serverConf.getString('host'),
+              port: serverConf.getNumber('port'),
+            } as InfinispanServerConfig),
+        );
+      } else if (typeof serversConfig === 'object' && serversConfig !== null) {
+        const serverConf = infinispanConfig.getConfig('servers');
+        parsedOptions.servers = {
+          host: serverConf.getString('host'),
+          port: serverConf.getNumber('port'),
+        } as InfinispanServerConfig;
+      } else {
+        logger?.error(
+          `Infinispan 'servers' configuration at ${storeConfigPath} must be an object or an array.`,
+        );
+        throw new Error(
+          `Infinispan 'servers' configuration at ${storeConfigPath} is invalid.`,
+        );
+      }
+    } else {
+      logger?.error(
+        `Infinispan configuration at ${storeConfigPath} is missing the 'servers' definition.`,
+      );
+      throw new Error(
+        `Infinispan configuration at ${storeConfigPath} must define 'servers'.`,
+      );
+    }
+
+    if (infinispanConfig.has('options')) {
+      const clientOptsConfig = infinispanConfig.getConfig('options');
+      const behaviorOptions: Partial<InfinispanClientBehaviorOptions> = {};
+
+      const versionStr = clientOptsConfig.getOptionalString('version');
+      if (
+        versionStr === '2.9' ||
+        versionStr === '2.5' ||
+        versionStr === '2.2'
+      ) {
+        behaviorOptions.version = versionStr; // Assign the string directly
+      } else if (versionStr === undefined || versionStr === null) {
+        behaviorOptions.version = versionStr; // Assign undefined or null
+      } else if (versionStr) {
+        // Log a warning if the version string is present but not one of the allowed values
+        logger?.warn(
+          `Invalid Infinispan client version "${versionStr}" in config at ${storeConfigPath}.options.version. Must be "2.9", "2.5", or "2.2". It will be ignored, and the client may use a default or fail.`,
+        );
+        // Do not assign behaviorOptions.version, let it be undefined if not valid
+      }
+
+      if (clientOptsConfig.has('cacheName'))
+        behaviorOptions.cacheName =
+          clientOptsConfig.getOptionalString('cacheName');
```

Comment:
you check that it exists, and then you get optionally - this seems redundant (applies to all of the steps below)